osmomath/decimal.go

```diff
@@ -248,6 +248,31 @@ func (d BigDec) BigInt() *big.Int {
 	return cp.Set(d.i)
 }
 
+// TruncateSDKInt() returns the sdkmath.Int representation of a BigDec.
+// Values in any additional decimal places are truncated.
+func (d BigDec) TruncateSDKInt() Int {
+	var precisionFactor *big.Int
+	precisionFactor = precisionFactors[0]
+	// Truncate any additional decimal values that exist due to BigDec's additional precision
+	// This relies on big.Int's Quo function doing floor division
+	intRepresentation := new(big.Int).Quo(d.BigInt(), precisionFactor)
+	// convert int representation back to SDK Int precision
+	truncatedInt := NewIntFromBigInt(intRepresentation)
+
+	return truncatedInt
+}
+
+// TruncateSDKIntMut() returns the sdkmath.Int representation of a BigDec,mutative the input.
+// Values in any additional decimal places are truncated.
+func (d BigDec) TruncateSDKIntMut() Int {
```

Comment:
If you need, I can push that Benchmark section

Reply:
True that there are perf wins, but still, internally, we're creating an internal variable of int representation and passing that as the argument for mutable int right?

Reply:
If the int is created internally, whats the difference of the two functions in the caller's point of view?

Reply:
With the TruncateToSDKIntMut() function, it will be used directly on the passed pointer d.i.

As for the TruncateToSDKInt() function, it will create a new pointer to avoid affecting the passed pointer d.i.

Reply:
TruncateToSDKIntMut() will optimize resources more